src/WebJobs.Extensions.DurableTask/DurableOrchestrationClient.cs

```diff
@@ -105,6 +106,22 @@ public override async Task<string> StartNewAsync(string orchestratorFunctionName
             return instance.InstanceId;
         }
 
+        private OrchestrationStatus[] GetStatusesNotToOverride()
+        {
+            var overridableStates = this.config.Options.OverridableExistingInstanceStates;
+            if (overridableStates == OverridableStates.NonRunningStates)
```

Comment:
I generally don't love using `if` statements with enums, instead I try to use `switch` statements. I find it a bit easier to tell when I have exhausted all of the enumerations at this point. In fact, there are even [some analyzers](https://roslyn-analyzers.readthedocs.io/en/latest/analyzers-info/enums/switch-on-enum-must-handle-all-cases.html#switch-on-enum-must-handle-all-cases) out in the wild that will give you warnings/exceptions if you add a new case to your enum without explicitly handling it in a switch statement.